gamemodes/guesswho/gamemode/sh_propfreeze.lua

```diff
@@ -0,0 +1,33 @@
+-- Needs few language functions to understand it better
+
+if SERVER then
+	local Enabled = GM.PropFreezeEnabled
+	local IsTimed = GM.PropFreezeIsTimed
+	local Delay = GM.PropFreezeDelay
+
+	function DoPropFreeze(ply)
+		for k, v in pairs( ents.FindByClass( "prop_*" ) ) do 
+			local phys = v:GetPhysicsObject()
+			if IsTimed == true and IsValid(phys) then
+				timer.Simple( Delay, function() if (IsValid(phys)) then phys:EnableMotion(false) end end )
+			elseif IsTimed == false and IsValid(phys) then
+				phys:EnableMotion(false)
+			end
+		end
+
+		local typemsg = "Freeze type set to"
+		if IsTimed == true then
+			print("[INTO] "..typemsg.." 'Timed'")
+		elseif IsTimed == false then
+			print("[INFO] "..typemsg.." 'Instant'")
+		end
+
+		print("[INFO] Props are frozen now!")
+	end
+
+	if Enabled == true then
```

Comment:
Always hook intoPosInit, And MapCleanup. In DoPropFreeze check if convar set with `GetConVar("gw_propfreeze_enabled"):GetBool()`.   So props will be frozen/unfrozen after a new round when convar changes. TL;DR `If convarforrozen then freeze else unfreeze end`